fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java

```diff
@@ -514,8 +534,12 @@ private boolean hasVocabularyConstraint(ElementDefinition elementDefinition) {
             String baseValueSet = (baseBinding != null) ? baseBinding.getValueSet().getValue() : null;
             String strength = binding.getStrength().getValue();
             String valueSet = binding.getValueSet().getValue();
-            return (!"required".equals(baseStrength) && "required".equals(strength))
-                    || ("required".equals(baseStrength) && "required".equals(strength) && !valueSetEqualsIgnoreVersion(valueSet, baseValueSet));
+            return (!"preferred".equals(baseStrength) && "preferred".equals(strength)) ||
+                    ("preferred".equals(baseStrength) && "preferred".equals(strength) && !valueSetEqualsIgnoreVersion(valueSet, baseValueSet)) ||
+                    (!"extensible".equals(baseStrength) && "extensible".equals(strength)) ||
+                    ("extensible".equals(baseStrength) && "extensible".equals(strength) && !valueSetEqualsIgnoreVersion(valueSet, baseValueSet)) ||
+                    (!"required".equals(baseStrength) && "required".equals(strength)) ||
+                    ("required".equals(baseStrength) && "required".equals(strength) && !valueSetEqualsIgnoreVersion(valueSet, baseValueSet));
```

Comment:
quite a piece of logic here, maybe a comment to explain?

Reply:
I have simplified this logic in the latest commit. It should be much easier to understand now.